hoot-core/src/main/cpp/hoot/core/io/HootApiDb.cpp

```diff
@@ -1645,7 +1645,7 @@ long HootApiDb::insertFolder(const QString& displayName, const long parentId, co
       " RETURNING id");
   }
   _insertFolder->bindValue(":displayName", displayName);
-  _insertFolder->bindValue(":parentId", (qlonglong)parentId);
+  _insertFolder->bindValue(":parentId", parentId == -1 ? QVariant() : (qlonglong)parentId);
```

Comment:
I believe that will work for passing a null QVariant. Not completely sure w/o trying. If that doesn't, then passing NULL might.

Reply:
It was complaining when i tried passing in just NULL. This solution seemed to make it null in the DB

Reply:
I think it used to allow NULL there in qt4 but doesn't any more in qt5...can't remember. So, it takes QVariant() ok? I wonder if not binding the val at all when parentId == -1 is an option.

Reply:
It looks like there is a slight difference between `QVariant()` and `QVariant(QVariant::Long)` but it may not affect anything here.  If @bobi7 is seeing a `NULL` value in the database then we are good to go.
https://doc.qt.io/qt-5/qvariant.html#QVariant-1

Reply:
Ok, so `parentId` here is a value from the C++ code, not a value read from the database.  I was confused because the addition of the foreign key we could not use `-1` to represent the root folder parent id.

Reply:
I tested out not binding the value when parentId == -1 and worked. I liked this approach more so just ended up using that